### PR TITLE
avro_options added along with test

### PR DESCRIPTION
**Pull request by sanghaniJ**
fixes https://github.com/hashicorp/terraform-provider-google/issues/10119

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
bigquery: added `avro_options` field to `google_bigquery_table` resource
```

### CODE REVIEW

File: mmv1/third_party/terraform/resources/resource_bigquery_table.go [hunk 1]
```diff
@@ -586,6 +586,25 @@ func resourceBigQueryTable() *schema.Resource {
...
+							Optional:    true,
+							MaxItems:    1,
+							Description: `Additional options if source_format is set to "AVRO"`,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"use_avro_logical_types": {
+										Type:        schema.TypeBool,
+										Optional:    true,
+										Description: `If sourceFormat is set to "AVRO", indicates whether to interpret logical types as the corresponding BigQuery data type (for example, TIMESTAMP), instead of using the raw type (for example, INTEGER).`,
+										AtLeastOneOf: []string{
```
Comment (megan07) [inline mmv1/third_party/terraform/resources/resource_bigquery_table.go:601]: We should actually make this required as there are no other options to be set in this block. If someone sets the `avro_options` block, we need a value set inside of it - and this is the only option right now.

File: mmv1/third_party/terraform/tests/resource_bigquery_table_test.go [hunk 2]
```diff
@@ -1555,6 +1581,44 @@ EOH
...
+  table_id   = "%s"
+  dataset_id = google_bigquery_dataset.test.dataset_id
+
+  external_data_configuration {
+    source_format = "AVRO"
+    autodetect = true
+    source_uris= ["gs://${google_storage_bucket.test.name}/*"]
+
+    avro_options {
+		use_avro_logical_types = true
```
Comment (megan07) [inline mmv1/third_party/terraform/tests/resource_bigquery_table_test.go:1613]: nit: the spacing seems to be inconsistent here

**Review by megan07 — CHANGES_REQUESTED** [review]
Thank you! Very good, just small changes. Thanks!

File: mmv1/third_party/terraform/resources/resource_bigquery_table.go (hunk 1 above)
Comment (sanghaniJ) [reply mmv1/third_party/terraform/resources/resource_bigquery_table.go]: I was going to do the same but as in the API documentation it was mentioned optional, I did the same but in different manner, but adding Required is a better way. So I've made the suggested changes.

**Pull request merged**